### PR TITLE
Disable otel warnings in shared_functions.hpp

### PR DESCRIPTION
**Pull request by LarryOsterman**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

File: sdk/eventhubs/azure-messaging-eventhubs/test/eventhubs-stress-test/src/scenarios/inc/shared_functions.hpp [hunk 1]
```diff
@@ -14,12 +14,25 @@
...
 #include <utility>
 
+#if defined(_MSC_VER)
+// The OpenTelemetry headers generate a couple of warnings on MSVC in the OTel 1.2 package, suppress
+// the warnings across the includes.
+#pragma warning(push)
+#pragma warning(disable : 4100)
+#pragma warning(disable : 4244)
+#pragma warning(disable : 6323) // Disable "Use of arithmetic operator on Boolean type" warning.
+#endif
```
Comment (ahsonkhan) [inline sdk/eventhubs/azure-messaging-eventhubs/test/eventhubs-stress-test/src/scenarios/inc/shared_functions.hpp:24]: Do we know of any other places that need these suppression?

Going forward, how do we reduce chances of re-introducing such errors, when new files get added? My assumption is all places that include `opentelemetry` headers need to have these warning guards aroudn them.

**Review by ahsonkhan — APPROVED** [review]
Are there any other live test pipelines we should run to validate this fixes all of them?

File: sdk/eventhubs/azure-messaging-eventhubs/test/eventhubs-stress-test/src/scenarios/inc/shared_functions.hpp (hunk 1 above)
Comment (ahsonkhan) [reply sdk/eventhubs/azure-messaging-eventhubs/test/eventhubs-stress-test/src/scenarios/inc/shared_functions.hpp:24]: Just did a quick search for `#include <opentelemetry/`, looks like only this file had these warning suppressions missing.

Comment (LarryOsterman) [reply sdk/eventhubs/azure-messaging-eventhubs/test/eventhubs-stress-test/src/scenarios/inc/shared_functions.hpp:24]: If there were, they wouldn't have passed the CI pipeline.

**Pull request merged**